src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -750,6 +750,45 @@ private static boolean isVCFStream(final InputStream stream, final String MAGIC_
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
+    private Object genotypeValsToTargetType(final String key, final String vals) {
+        if (!header.hasFormatLine(key)) {
+            return vals;
+        }
+
+        final VCFFormatHeaderLine formLine = header.getFormatHeaderLine(key);
+        final VCFHeaderLineType valType = formLine.getType();
+        final boolean isSingleVal = !vals.contains(VCFConstants.GENOTYPE_ATTRIBUTE_VALUE_SEPARATOR);
+
+
+        return isSingleVal
+                ? getMapperFunction(valType).apply(vals)
+                : mapToFormatLineType(
+                        ParsingUtils.split(vals, VCFConstants.GENOTYPE_ATTRIBUTE_VALUE_SEPARATOR_CHAR),
+                        valType);
+    }
+
+    private List<?> mapToFormatLineType(final List<String> genVals, final VCFHeaderLineType genValType) {
+        final List<Object> genRealTypeVals = new ArrayList<>(genVals.size());
+        final Function<String, Object> mapperFunc = getMapperFunction(genValType);
+
+        genVals.forEach(genVal -> genRealTypeVals.add(
```

Comment:
these should catch parsing exceptions and convert them to TribbleException explaining what happened